### PR TITLE
Add "View" button to current site card

### PR DESCRIPTION
**Pull request by sirreal**
This PR addresses #12781. 

The purpose is to clarify that clicking on the site will take you to the site.

To test:

1. Open a site ( `https://wordpress.com/stats/insights/$site` )
1. Check that the appearance is coherent and that site card the link functions correctly (see screenshots).
1. Verify that `homeLink=false` site cards _do not_ display the button (`/sites` or by clicking "Switch site" in the sidebar)

### CODE REVIEW

**Comment by sirreal** [comment]
## New button
view
## Hover state
view hover

**Comment by sirreal** [comment]
cc @rickybanister

**Comment by sirreal** [comment]
cc @jeherve 
Related Jetpack Masterbar issue: https://github.com/Automattic/jetpack/issues/6877

**Comment by mtias** [comment]
Can we use the "home" gridicon instead of text?

File: client/blocks/site/style.scss [hunk 1]
```diff
@@ -118,6 +119,39 @@
...
+	flex: 0 1 auto;
+	position: relative;
+
+	// from layout/sidebar/styles.scss a.sidebar-button
+	box-sizing: border-box;
+	white-space: nowrap;
+	overflow: hidden;
+	padding: 2px 8px 3px 8px;
+	height: 24px;
+	margin: 11px 8px 0 0;
```
Comment (mtias) [inline client/blocks/site/style.scss]: are you using top margin to position things? `align-items: center` from the flex parent should do it.

File: client/blocks/site/style.scss [hunk 2]
```diff
@@ -118,6 +119,39 @@
...
+	line-height: 18px;
+	background-color: $gray-light;
+	color: darken( $gray, 20% );
+	font-size: 11px;
+	font-weight: 600;
+	border-radius: 3px;
+	border: 1px solid darken( $sidebar-bg-color, 10% );
```
Comment (mtias) [inline client/blocks/site/style.scss]: Shouldn't use sidebar colors for this item because it can live outside of the sidebar.

**Comment by sirreal** [comment]
@mtias 
> Can we use the "home" gridicon instead of text?

I'm happy to try with the home icon. We will have home icons on both sides in hover state, so the hover icon on the left would need to change. Do you have any ideas for that?

@folletto might care to chime in:
> Adding an icon doesn't seem to me that more explicit than the current situation.
https://github.com/Automattic/wp-calypso/issues/12781#issuecomment-292490544

**Comment by mtias** [comment]
@sirreal I'd expect we drop the hover home icon with this change regardless of how the button looks.

**Comment by folletto** [comment]
Sorry if take a step back, but I feel the discussion in #12781 didn't reach a point yet where we have a solution to build. I'm ok in having this as a way to prototype it, but I wouldn't proceed yet.

This specifically:

1. Text can be too verbose, especially translated
2. Where would the update icon fit (Jetpack)? It's in that position.

Neither icon, nor text, feels a clear improvement to me over what we have now. Can we keep the discussion going before we ship?

File: client/blocks/site/style.scss (hunk 1 above)
Comment (sirreal) [inline client/blocks/site/style.scss]: Much cleaner, thanks!

**Comment by sirreal** [comment]
@folletto Its not my intention to rush things, just getting the ball rolling. I know there's design discussion that needs to occur.

> Where would the update icon fit (Jetpack)? It's in that position.
Good question. The indicator is in the same position, but I don't know if there is use overlap. I've only seen the indicator when switching sites and I've only seen the home link when it's at the top of the sidebar and you're viewing a site.

Regardless, it would be good to avoid showing both icons simultaneously. The simple choice is just to prefer one over the other. It seems reasonable to prefer the "home" icon over the update indicator because in that state clicking should open the site.

Hover is much more subtle with the icon.

home
home-hover

**Comment by johnHackworth** [comment]
> I'm ok in having this as a way to prototype it, but I wouldn't proceed yet.

@folletto would you be ok as in a first step we roll this change just for jetpack sites while we discuss something more permanent for everyone? I'm under the impression that they need this link way more heavily than regular dot com sites

**Comment by folletto** [comment]
We should also strive to make Jetpack identical to .com, not add more differences.

Even more, it seems far more important for Jetpack sites that the update badge not appearing #10832 bug is fixed — and that takes the same position of the icon/button discussed here.

**Comment by ashleighaxios** [comment]
This solution doesn't quite work. It's likely worth waiting and rolling something out that can work for all site uses, keeping more consistency.

I think we can continue iterating on the design. The patterns we have tend not to have a button when the whole element is clickable, so we should keep that interaction standard. I also agree that just the home icon changing on hover/touch may still be too subtle without the button. So, from there, there can be some quick design iterations and then this can be picked back up. 

It seems like @rickybanister is leading design on this one (? if not, let's make sure there's a clear point person). So we should wait for him to chime back in when he's online.

_I still think this can happen fast. Just don't ship yet :)_

**Comment by johnHackworth** [comment]
> We should also strive to make Jetpack identical to .com, not add more differences.

Agree... but that would mean that we would need to cover jetpack sites requirements in calypso too.

> Even more, it seems far more important for Jetpack sites

Hmm, I don't know if I agree. It looks like one the biggest pains right now for jetpack users in calypso is that is not so obvious how to return to their own site without manually typing their URL. Which is something that jetpack users want to do, often (there are several flows that guide them to calypso, so they can find themselves there without a clear way to go back to their site). In a future / ideal world, they won't need to 'come back' to wp-admin ... but the thing is that right now they need a way to do it (especially because sometimes they are not coming to calypso willingly).

So I think that even if I wholeheartedly agree with "jetpack experience should be identical to dot com", I think we need to contemplate those users necessities while we can offer a better alternative

**Comment by folletto** [comment]
The pain point exists for everyone, not just Jetpack, which is why "going back to the site" is probably one of the most iterated pieces in our sidebar design. 

Jetpack however is the only one that _needs_ that space for the update badge, and the bug is still open and unfixed – a need not present on .com sites.

**Comment by johnHackworth** [comment]
Yeah, my point is that dot com users are in calypso willingly. They know where they are and what they are doing there... Jetpack users, in other hand... well, not always. So not offering a visible and immediate way to go back sounds way worse in their cases :)

**Comment by rickybanister** [comment]
How about going with something like this (I believe @shaunandrews proposed this before):

image

Yes, it will add another item to the sidebar, but I think we're due for some pruning there anyway. The primary front-to-back navigation to your site seems more important than some other items.

Also we will be moving /sharing into /settings very soon opening up a slot. 

Also, @folletto the site indicator bug needs to be fixed, but I don't like the dot being present when that site is selected. I think the dot is useful in the sites list, but once you have selected a site I would rather we show the compact notice below the site card as in @sirreal's screenshot above. I don't think we should require an extra click to reveal the problem with the site, it should be easily accessible.

**Comment by rickybanister** [comment]
Just one other note, I originally proposed the 'visit' button because it mirrored the other sidebar buttons and was explicitly buttony. I didn't mean for the entire card to be clickable. I also didn't want to go with just an icon floating to the right of the card because that's not explicitly buttony or clickable. There are already instances of other icons appearing in that card (the lock for private sites, etc) so it is not obvious that the existence of an icon means it is the access point to your site. 

I did mean for the home icon to go away on hover, however. 

What do people think of an explicit sidebar link instead? It won't impede the site indicator thing, it won't add more to the sidebar because we are freeing up a space very soon, it matches the sidebar pattern for the most part, etc etc.

**Comment by folletto** [comment]
> How about going with something like this 

I'm good with this... as it was the original Calypso v1 design! :D 
I think it's still the clearest option, and on the front-end it would be highlighted, so it's even clearer.

> I think we're due for some pruning there anyway

Menus is about to be removed, fwiw.

**Comment by rickybanister** [comment]
yay, a two-for-one deal!

**Comment by mtias** [comment]
You are only allowed to add an item if you remove one. (Menus don't count.) ;)

My biggest dislike with "View site" as a sidebar item is that it doesn't fit in the hierarchy. What about a "view site" in the sidebar footer instead?

**Comment by rickybanister** [comment]
I don't think a footer item would have the right 'flow'. It being the top item feels right, maybe there's a typographic thing we could change about it?

**Comment by sirreal** [comment]
I've created #13176 to implement the different direction the design has taken. Please comment there.

**Pull request closed without merging**